### ISSUE TITLE
"[query_string] query does not support [fields]" error

### ISSUE THREAD

**Issue by gpetrou** [elastic/elasticsearch-net#8036]
**Elastic.Clients.Elasticsearch version**: 8.12.0

**Elasticsearch version**: 8.2.2

**.NET runtime version**: net8.0

**Operating system version**: Windows 11

**Description of the problem including expected versus actual behavior**:
Using Fields property of QueryStringQuery does not work anymore. There is a "[query_string] query does not support [fields]" error.

**Steps to reproduce**:
```
Elastic.Clients.Elasticsearch.QueryDsl.QueryStringQuery queryStringQuery = new()
{
    Fields = new[] { "test" }
};

SearchRequest<JsonObject> elasticSearchRequest = new("IndexName")
{
    Query = queryStringQuery
};

Elastic.Clients.Elasticsearch.SearchResponse<JsonObject> expected = await _elasticsearchClient.SearchAsync<JsonObject>(elasticSearchRequest);
```

**Expected behavior**
The code does not result in an error.

**Provide `ConnectionSettings` (if relevant)**:

**Provide `DebugInformation` (if relevant)**:

**Comment by CodeDan90** [comment]
Currently having the same problem with version 8.12.0, when you only use one field the error occurs but if you add more than one field the query works.

**Comment by gpetrou** [comment]
@stevejgordon perhaps https://github.com/elastic/elasticsearch-net/pull/8018 broke this? In that PR, I don't see any tests. Are there tests in another repository that verified this change?

**Comment by stevejgordon** [comment]
@gpetrou Yes, unfortunately, that change which fixes `Fields` usage elsewhere has caused this issue. @flobernd, it looks like this will need to be addressed in the specification, as it seems there are different behaviours at play in ES for single-field entries. We may need to model both scenarios.

**Comment by gpetrou** [comment]
@flobernd @stevejgordon is there a plan to fix this bug? Should we expect a hotfix?

**Comment by flobernd** [comment]
@gpetrou Yes

**Comment by gpetrou** [comment]
> Yes. In the meantime please downgrade to the last version, if possible.

We were never able to upgrade due to this issue. Is this going to be addressed in the upcoming release?

**Comment by flobernd** [comment]
Yes. This will be fixed in 8.12.1 by reverting the PR that broke it.

In one of the 8.13.x patch releases I will reintroduce the (stored_fields) fix in a way that does not affect the QueryStringQuery.

**Comment by flobernd** [comment]
Fixed in 8.12.1
